x-pack/plugins/ml/public/components/job_selector/job_selector.js

```diff
@@ -0,0 +1,457 @@
+            {i18n.translate('xpack.ml.jobSelector.showBarBadges', {
+              defaultMessage: `And {overFlow} more`,
+              values: { overFlow },
+            })}
+          </EuiText>
+        </EuiLink>);
+
+      return badges;
+    }
+  }
+
+  function renderNewSelectionIdBadges() {
+    const badges = [];
+
+    for (let i = 0; i < newSelection.length; i++) {
+      if (i >= BADGE_LIMIT && showAllBadges === false) {
+        break;
+      }
+
+      badges.push(
+        <EuiFlexItem grow={false} key={newSelection[i]}>
+          {getBadge({
+            id: newSelection[i],
+            icon: true,
+            removeId,
+            isGroup: (maps.groupsMap[newSelection[i]] !== undefined)
+          })}
+        </EuiFlexItem>
+      );
+    }
+
+    if (showAllBadges === false && newSelection.length > BADGE_LIMIT) {
+      badges.push(
+        <EuiLink
+          key="more-badges-link"
+          onClick={() => setShowAllBadges(!showAllBadges)}
+        >
+          <EuiText grow={false} size="xs">
+            {i18n.translate('xpack.ml.jobSelector.showFlyoutBadges', {
+              defaultMessage: `And {overFlow} more`,
+              values: { overFlow: newSelection.length - BADGE_LIMIT },
+            })}
+          </EuiText>
+        </EuiLink>);
+    } else if (showAllBadges === true && newSelection.length > BADGE_LIMIT) {
+      badges.push(
+        <EuiLink
+          key="hide-badges-link"
+          onClick={() => setShowAllBadges(!showAllBadges)}
+        >
+          <EuiText grow={false} size="xs">
+            {i18n.translate('xpack.ml.jobSelector.hideFlyoutBadges', {
+              defaultMessage: 'Hide'
+            })}
+          </EuiText>
+        </EuiLink>);
+    }
+
+    return badges;
+  }
+
+  function renderJobSelectionBar() {
+    return (
+      <EuiFlexGroup responsive={false} gutterSize="xs" alignItems="center">
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty
+            onClick={handleJobSelectionClick}
+          >
+            {i18n.translate('xpack.ml.jobSelector.jobSelectionButton', {
+              defaultMessage: 'Job Selection'
+            })}
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup wrap responsive={false} gutterSize="xs" alignItems="center">
+            {renderIdBadges()}
+          </EuiFlexGroup>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  function renderFlyout() {
+    if (isFlyoutVisible) {
+      return (
+        <EuiFlyout
+          onClose={closeFlyout}
+          aria-labelledby="Job Selection"
+          size="l"
+        >
+          <EuiFlyoutHeader
+            hasBorder
+            className="mlJobSelectorFlyoutHeader"
+          >
+            <EuiTitle size="m">
+              <h2 id="flyoutTitle">
+                {i18n.translate('xpack.ml.jobSelector.flyoutTitle', {
+                  defaultMessage: 'Job Selection'
```

Comment:
Should probably use sentence case `Job selection` here according to the EUI writing guidelines.